alacritty/src/window.rs

```diff
@@ -374,17 +393,22 @@ impl Window {
         self.window().set_simple_fullscreen(simple_fullscreen);
     }
 
-    #[cfg(not(any(target_os = "macos", target_os = "windows")))]
-    pub fn wayland_display(&self) -> Option<*mut c_void> {
+    #[cfg(any(not(feature = "wayland"), any(target_os = "macos", windows)))]
+    pub fn wayland_display(&self) -> Option<*mut std::ffi::c_void> {
+        None
+    }
+
+    #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+    pub fn wayland_display(&self) -> Option<*mut std::ffi::c_void> {
         self.window().wayland_display()
     }
```

Comment:
Could we order these methods uniformly to have the simple "does nothing" version always at the end? I think that makes things a bit easier to understand when reading through things from top to bottom.